tests/unit_tests/scheduler/conftest.py

```diff
@@ -10,19 +10,26 @@ def __init__(self, init=None, wait=None, kill=None):
         self._mock_wait = wait
         self._mock_kill = kill
 
-    async def _init(self, *args, **kwargs):
+    async def _init(self, iens, *args, **kwargs):
         if self._mock_init is not None:
-            await self._mock_init(*args, **kwargs)
+            await self._mock_init(iens, *args, **kwargs)
+        return iens
 
-    async def _wait(self, *args):
+    async def _wait(self, iens):
         if self._mock_wait is not None:
-            result = await self._mock_wait()
+            if self._mock_wait.__code__.co_argcount > 0:
+                result = await self._mock_wait(iens)
+            else:
+                result = await self._mock_wait()
             return True if result is None else bool(result)
         return True
 
-    async def _kill(self, *args):
+    async def _kill(self, iens, *args):
```

Comment:
`_kill` takes no more arguments

```suggestion
    async def _kill(self, iens):
```